pandas/core/groupby/groupby.py

```diff
@@ -3219,6 +3226,55 @@ def max(
                 npfunc=np.max,
             )
 
+    def _idxmax_idxmin(
+        self,
+        how: Literal["idxmin", "idxmax"],
+        alt: Callable,
+        numeric_only: bool = False,
+    ) -> Series | DataFrame:
+        """Compute idxmax/idxmin.
+
+        Parameters
+        ----------
+        how: {"idxmin", "idxmax"}
+            Whether to compute idxmin or idxmax.
+        alt: Callable
+            Fallback for non-Cython (e.g. object) types.
+        numeric_only : bool, default False
+            Include only float, int, boolean columns.
+
+        Returns
+        -------
+        Series or DataFrame
+            idxmax or idxmin for the groupby operation.
+        """
+
+        def post_process(x):
+            if x.size == 0:
+                result = x.astype(self.obj.index.dtype)
+            else:
+                index = self.obj.index
+                if isinstance(index, MultiIndex):
+                    index = index.to_flat_index()
+                result = index.take(x).values
```

Comment:
could be simpler to do `index.array.take(x, allow_fill=True)`?

Reply:
For most index types, this attempts to call `_take_nd_ndarray` which then calls `_libs.algos.take_1d_int64_float64` and  fails since `x` here is (often) 2d.